### PR TITLE
(rpc_server): Extend metrics

### PR DESCRIPTION
**Pull request by kobayurii**
In this PR Historical/regular counter is extended and broken down by method.
This metric should to help to analyze the block_references for methods 
to see the most common pattern is it specific block or final/optimistic.

- final
- optimistic
- block_id:
     - regular - includes 5 last epochs
     - historical - later than 5 epochs ago

### CODE REVIEW

File: rpc-server/src/metrics.rs [hunk 1]
```diff
@@ -96,12 +96,18 @@ lazy_static! {
...
 
-    pub(crate) static ref REQUESTS_COUNTER: IntCounterVec = register_int_counter_vec(
-        "requests_counter",
+    pub(crate) static ref TOTAL_REQUESTS_COUNTER: IntCounterVec = register_int_counter_vec(
+        "total_requests_counter",
         "Total number of requests",
         &["request_type"] // This declares a label named `request_type`
     ).unwrap();
 
+    pub(crate) static ref TIMELINE_METHOD_REQUESTS_COUNTER: IntCounterVec = register_int_counter_vec(
```
Comment (khorolets) [inline rpc-server/src/metrics.rs:105]: I think it'd be better to merge these two metrics `TOTAL_REQUESTS_COUNTER` and `TIMELINE_METHOD_REQUESTS_COUNTER` into one.

My assumption is that using Grafana we will be able to sum up the broken down metrics into a single value (if needed).

What do you think? Is there any benefit of having two similar yet different in details metrics?

File: database/src/base/rpc_server.rs [hunk 2]
```diff
@@ -94,19 +94,19 @@ pub trait ReaderDbManager {
         &self,
         transaction_hash: &str,
         method_name: &str,
-    ) -> anyhow::Result<readnode_primitives::TransactionDetails>;
+    ) -> anyhow::Result<(u64, readnode_primitives::TransactionDetails)>;
```
Comment (khorolets) [inline database/src/base/rpc_server.rs:97]: I agree, sometimes there might be a situation where returning a tuple is inevitable. Although, I don't believe this is such a situation.

Please consider introducing something like

```rust
struct ResponseWithBlockHeight<T> {
  block_height: u64,
  response: T,
}
```

Thus it'll be more maintainable in the future and less error-prone.

**Review by khorolets — CHANGES_REQUESTED** [review]
Overall, the metric improvements look good. I've highlighted a few places I think we'd benefit from.

File: database/src/base/rpc_server.rs (hunk 2 above)
Comment (kobayurii) [reply database/src/base/rpc_server.rs:98]: We expected to rewrite the database crate. I left a TODO comment for the future.

**Review by khorolets — APPROVED** [review]

**Pull request merged**